packages/frontend/core/src/modules/explorer/views/tree/node.tsx

```diff
@@ -117,6 +120,27 @@ export const ExplorerTreeNode = ({
   const [renaming, setRenaming] = useState(defaultRenaming);
   const [lastInGroup, setLastInGroup] = useState(false);
   const rootRef = useRef<HTMLDivElement>(null);
+  const { emoji, name } = useMemo(() => {
+    if (!extractEmojiAsIcon || !rawName) {
+      return {
+        emoji: null,
+        name: rawName,
+      };
+    }
+    const isStartsWithEmoji = /^(\p{Emoji_Presentation})/u.test(rawName);
```

Comment:
may extract this to a utility function? I guess it might be reused in places like in tab header
![image](https://github.com/user-attachments/assets/dbdc6a28-1392-4034-9318-46fbcd0cbbd2)